src/main/java/net/neoforged/neoforge/client/settings/KeyModifier.java

```diff
@@ -90,10 +128,16 @@ public boolean isActive(@Nullable IKeyConflictContext conflictContext) {
         public Component getCombinedName(InputConstants.Key key, Supplier<Component> defaultLogic) {
             return defaultLogic.get();
         }
+
+        @Override
+        public InputConstants.Key[] codes() {
+            return KEYS;
+        }
     };
 
     public static final KeyModifier[] MODIFIER_VALUES = { SHIFT, CONTROL, ALT };
 
+    @Deprecated(forRemoval = true, since = "1.21.4")
```

Comment:
Ideally, this should have a `/** @deprecated */` comment pointing to the new method.